pkg/apis/awsprovider/v1alpha1/types.go

```diff
@@ -102,6 +102,12 @@ type Network struct {
 	APIServerELB ClassicELB `json:"apiServerElb,omitempty"`
 }
 
+// ClusterControlPlaneStatus encapsulates control-plane status for the cluster
+type ClusterControlPlaneStatus struct {
+	// ControlPlaneNodeCount is the count of the number of control-plane nodes in the cluster.
+	ControlPlaneNodeCount int `json:"controlPlaneNodeCount"`
```

Comment:
Should this just be `NodeCount` given that the struct already sets the context?

Reply:
Dropping this approach